mlir/lib/CAPI/Dialect/Rock.cpp

```diff
@@ -0,0 +1,342 @@
+//===- MIGraphX.cpp - C Interface for MIGraphX dialect
+//------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir-c/Dialect/Rock.h"
+#include "mlir/CAPI/IR.h"
+#include "mlir/CAPI/Pass.h"
+#include "mlir/CAPI/Registration.h"
+#include "mlir/CAPI/Wrap.h"
+#include "mlir/Dialect/Rock/IR/Rock.h"
+#include "mlir/Dialect/Rock/Tuning/ConvContext.h"
+#include "mlir/Dialect/Rock/Tuning/RockTuningTypes.h"
+
+#include <vector>
+
+MLIR_DEFINE_CAPI_DIALECT_REGISTRATION(Rock, rock, mlir::rock::RockDialect)
+DEFINE_C_API_PTR_METHODS(MlirRockTuningSpace, mlir::rock::TunableParams)
+DEFINE_C_API_PTR_METHODS(MlirRockTuningParam, mlir::rock::ParamEntry)
+// DEFINE_C_API_METHOD(MlirRockTuningKey, const void);
+
+using namespace mlir;
+
+// Brute-force search in incremental order
+void createConv2DFwdTuningRangeBF(struct rock::TunableParams *newSpace,
+                                  rock::Conv2DOp convOp) {
+
+  auto bXdlops = convOp->getAttrOfType<BoolAttr>("xdlopsV2");
+  if (bXdlops != nullptr && bXdlops.getValue() == true) {
+    // XDLOPS
+    // M/block N/block K/block M/wave N/wave kPack aCopyMore bCopyMore
+    std::vector<std::vector<uint32_t>> tParams = {{4, 8, 16, 32, 64, 128},
```

Comment:
If these are always constant, let's make it a constexpr so it is only built at compile time.